src/drivers/uavcan/sensors/baro.cpp

```diff
@@ -83,22 +87,40 @@ void
 UavcanBarometerBridge::air_pressure_sub_cb(const
 		uavcan::ReceivedDataStructure<uavcan::equipment::air_data::StaticPressure> &msg)
 {
-	sensor_baro_s report{};
-
-	/*
-	 * FIXME HACK
-	 * This code used to rely on msg.getMonotonicTimestamp().toUSec() instead of HRT.
-	 * It stopped working when the time sync feature has been introduced, because it caused libuavcan
-	 * to use an independent time source (based on hardware TIM5) instead of HRT.
-	 * The proper solution is to be developed.
-	 */
-	report.timestamp   = hrt_absolute_time();
-	report.temperature = last_temperature_kelvin - 273.15F;
-	report.pressure    = msg.static_pressure / 100.0F;  // Convert to millibar
-	report.error_count = 0;
-
-	/* TODO get device ID for sensor */
-	report.device_id = 0;
-
-	publish(msg.getSrcNodeID().get(), &report);
+	lock();
+	uavcan_bridge::Channel *channel = get_channel_for_node(msg.getSrcNodeID().get());
+	unlock();
+
+	if (channel == nullptr) {
+		// Something went wrong - no channel to publish on; return
+		return;
+	}
+
+	// Cast our generic CDev pointer to the sensor-specific driver class
+	PX4Barometer *_baro = (PX4Barometer *)channel->h_driver;
+
+	_baro->set_temperature(last_temperature_kelvin - 273.15f);
+	_baro->update(hrt_absolute_time(), msg.static_pressure / 100.0f); // Convert pressure to millibar
+}
+
+int UavcanBarometerBridge::init_driver(uavcan_bridge::Channel *channel)
+{
+	// update device id as we now know our device node_id
+	DeviceId device_id{_device_id};
+
+	// No sensor info is included int he StaticPressure msg; use some generic baro type
+	device_id.devid_s.devtype = DRV_BARO_DEVTYPE_MS5611;
```

Comment:
I'm wondering if we should have a generic option or similar for UAVCAN.

Reply:
Ideally this would be included in the UAVCAN message; in lieu of that, I'm open to suggestions.  The fact that it's UAVCAN is actually already accounted for in the "bus type" field:

`_device_id.devid_s.bus_type = DeviceBusType_UAVCAN;`

But adding a separate, not-currently-assigned value for a generic UAVCAN mag/baro/etc would be a step in the right direction.

Reply:
Forgot to mention, I added a section of UAVCAN device types to the list (0x80 to 0x89 range).